test/test-PerfMeter.js

```diff
@@ -0,0 +1,196 @@
+/* globals Services */
+
+"use strict";
+const simplePrefs = require("sdk/simple-prefs");
+
+const {before, after} = require("sdk/test/utils");
+const tabs = require("sdk/tabs");
+const {ActivityStreams} = require("lib/ActivityStreams");
+const {Cu} = require("chrome");
+Cu.import("resource://gre/modules/Services.jsm");
+
+let app;
+
+function openTestTab(openUrl, notifyEvent) {
+  return new Promise(resolve => {
+    let tabData = {};
+    function onShow(tab) {
+      tabData.tab = tab;
+      if (!notifyEvent) {
+        resolve(tabData);
+      }
+    }
+
+    let observer = {
+      observe(subject, topic, data) {
+        Services.obs.removeObserver(this, notifyEvent);
+        tabData.notifyData = JSON.parse(data);
+        resolve(tabData);
+      }
+    };
+
+    if (notifyEvent) {
+      Services.obs.addObserver(observer, notifyEvent);
+    }
+
+    // open the page in timeout, to make it fire after we return
+    tabs.open({url: openUrl, onPageShow: onShow});
+  });
+}
+
+function prefSetter(value) {
+  return new Promise(resolve => {
+    let observer = {
+      observe() {
+        Services.obs.removeObserver(this, "performance-pref-changed");
+        resolve();
+      }
+    };
+    Services.obs.addObserver(observer, "performance-pref-changed");
+    simplePrefs.prefs.performanceLog = value;
+  });
+}
+
+function verifyPerfEvents(eventsArray, assert) {
+  let exectedEvents = new Set([
+    "TAB_OPEN",
+    "WORKER_ATTACHED",
+    "TAB_READY",
+    "TOP_FRECENT_SITES_REQUEST",
+    "RECENT_LINKS_REQUEST",
+    "FRECENT_LINKS_REQUEST",
+    "RECENT_BOOKMARKS_REQUEST",
+    "SEARCH_STATE_REQUEST",
+    "NOTIFY_PERFORMANCE",
+    "TOP_FRECENT_SITES_RESPONSE",
+    "RECENT_LINKS_RESPONSE",
+    "FRECENT_LINKS_RESPONSE",
+    "RECENT_BOOKMARKS_RESPONSE",
+    "SEARCH_STATE_RESPONSE",
+    "NOTIFY_PERFORMANCE",
+  ]);
+
+  let seenEvents = new Set();
+  for (let event of eventsArray) {
+    assert.ok(exectedEvents.has(event.tag), "Received expected event " + event.tag);
+    assert.ok(!seenEvents.has(event.tag + event.data), "Event tag and data pair is unique");
+    if (event.tag === "NOTIFY_PERFORMANCE") {
```

Comment:
why not add `NOTIFY_PERFORMANCE`, `BASE_MOUNTED` and `NEWTAB_RENDER` in `expectedEvents?